scheduler/plugin/wrappedplugin.go

```diff
@@ -0,0 +1,270 @@
+package plugin
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+
+	schedulingresultstore "github.com/kubernetes-sigs/kube-scheduler-simulator/scheduler/plugin/resultstore"
+)
+
+//go:generate mockgen -destination=./mock/$GOFILE -package=plugin . Store,FilterPluginExtender,ScorePluginExtender,NormalizeScorePluginExtender
+//go:generate mockgen -destination=./mock/framework.go -package=plugin k8s.io/kubernetes/pkg/scheduler/framework FilterPlugin,ScorePlugin,ScoreExtensions
+type Store interface {
+	AddNormalizedScoreResult(namespace, podName, nodeName, pluginName string, normalizedscore int64)
+	AddFilterResult(namespace, podName, nodeName, pluginName, reason string)
+	AddScoreResult(namespace, podName, nodeName, pluginName string, score int64)
+}
+
+type FilterPluginExtender interface {
+	// BeforeFilter is a function that runs before the Filter method of the original plugin.
+	// If BeforeFilter returns non-success status, the simulator plugin doesn't run the Filter method of the original plugin and return that status.
+	BeforeFilter(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodeInfo *framework.NodeInfo) *framework.Status
+	// AfterFilter is a function that runs after the Filter method of the original plugin.
+	// A Filter of the simulator plugin finally returns the status returned from AfterFilter.
```

Comment:
Please change other AfterXXXX comments as well
```suggestion
	// AfterFilter is a function that is run after the Filter method of the original plugin.
	// AfterFilter must be run regardless of the result from a Filter method of the original plugin,
	// and a Filter method of the simulator plugin returns the status returned from AfterFilter.
```